jupyter notebook magic: is it possible to hide warnings generated by Pythran?

Hi @serge-sans-paille,

thanks for continuing to put a lot of work into Pythran.
I'm mostly using Pythran from the Jupyter Notebook, using the nice %%pythran.magic command.
However, this always leads to a lot of output that I don't want to see in my notebook, that is, mostly warnings from the compiler. 
An example here can be found here on my blog: [http://flothesof.github.io/optimizing-python-code-numpy-cython-pythran-numba.html](http://flothesof.github.io/optimizing-python-code-numpy-cython-pythran-numba.html).
The output is full with stuff like: 
```
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp89u4ysag.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:12:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/utils/broadcast_copy.hpp:4:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp:63:5: warning: struct template 'numpy_gexpr' was previously declared as a class template [-Wmismatched-tags]
    struct numpy_gexpr;
    ^
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/utils/numpy_traits.hpp:19:11: note: previous use is here
    class numpy_gexpr;
          ^
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp89u4ysag.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:12:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/utils/broadcast_copy.hpp:4:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp:373:3: warning: 'tuple_element' defined as a struct template here but previously declared as a class template [-Wmismatched-tags]
  struct tuple_element<I, pythonic::types::array<T, N>> {
  ^
/Library/Developer/CommandLineTools/usr/include/c++/v1/__tuple:57:34: note: did you mean struct here?
template <size_t _Ip, class _Tp> class _LIBCPP_TEMPLATE_VIS tuple_element;
                                 ^
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp89u4ysag.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:12:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/utils/broadcast_copy.hpp:4:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/tuple.hpp:378:3: warning: 'tuple_size' defined as a struct template here but previously declared as a class template [-Wmismatched-tags]
  struct tuple_size<pythonic::types::array<T, N>> {
  ^
/Library/Developer/CommandLineTools/usr/include/c++/v1/__tuple:25:22: note: did you mean struct here?
template <class _Tp> class _LIBCPP_TEMPLATE_VIS tuple_size;
                     ^
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp89u4ysag.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:16:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/list.hpp:12:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/vectorizable_type.hpp:69:5: warning: struct template 'numpy_expr' was previously declared as a class template [-Wmismatched-tags]
    struct numpy_expr;
    ^
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/utils/numpy_traits.hpp:28:11: note: previous use is here
    class numpy_expr;
          ^
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp89u4ysag.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:16:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/list.hpp:44:5: warning: struct template 'ndarray' was previously declared as a class template [-Wmismatched-tags]
    struct ndarray;
    ^
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/utils/numpy_traits.hpp:10:11: note: previous use is here
    class ndarray;
          ^
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp89u4ysag.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/ndarray.hpp:16:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/types/list.hpp:380:3: warning: 'tuple_element' defined as a struct template here but previously declared as a class template [-Wmismatched-tags]
  struct tuple_element<I, pythonic::types::list<T>> {
  ^
/Library/Developer/CommandLineTools/usr/include/c++/v1/__tuple:57:34: note: did you mean struct here?
template <size_t _Ip, class _Tp> class _LIBCPP_TEMPLATE_VIS tuple_element;
```

Is it possible to add a flag to the Magic or even better: hide the output by default? I believe the output is necessary when there is an error, but no so when it just works. 

What do you think? I've tried to look at the implementation of the magic, but unfortunately I'm not able to do the required changes on my own. I'd love to contribute fixing this, if you can push me in the right direction.

Thanks,
Florian

Hi,

thanks for being so quick.
I checked out your branch with the fix, created a clean Python 3.6 environment for testing and ran the CLI again. 
I still get warnings. 
When I run the pythran command with the -v flag, I get the following output:
```
ARNING: Disabling color, you really want to install colorlog.
WARNING: Pythran support disabled for module: omp
INFO: customize UnixCCompiler
running build_ext
running build_src
build_src
building extension "a" sources
build_src: building npy-pkg config files
new_compiler returns <class 'distutils.unixccompiler.UnixCCompiler'>
INFO: customize UnixCCompiler
customize UnixCCompiler using build_ext
********************************************************************************
<class 'distutils.unixccompiler.UnixCCompiler'>
preprocessor  = ['gcc', '-E']
compiler      = ['gcc', '-Wno-unused-result', '-Wsign-compare', '-Wunreachable-code', '-DNDEBUG', '-g', '-fwrapv', '-O3', '-Wall', '-Wstrict-prototypes', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64']
compiler_so   = ['gcc', '-Wno-unused-result', '-Wsign-compare', '-Wunreachable-code', '-DNDEBUG', '-g', '-fwrapv', '-O3', '-Wall', '-Wstrict-prototypes', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64']
compiler_cxx  = ['g++']
linker_so     = ['gcc', '-bundle', '-undefined', 'dynamic_lookup', '-Wl,-rpath,/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-headerpad_max_install_names', '-arch', 'x86_64', '-Wl,-rpath,/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-headerpad_max_install_names', '-arch', 'x86_64']
linker_exe    = ['gcc']
archiver      = ['ar', 'rc']
ranlib        = ['ranlib']
libraries     = []
library_dirs  = ['/Users/kappamaki/anaconda/lib']
include_dirs  = ['/Users/kappamaki/anaconda/include/python3.6m']
********************************************************************************
new_compiler returns <class 'distutils.unixccompiler.UnixCCompiler'>
INFO: customize UnixCCompiler
customize UnixCCompiler using build_ext
********************************************************************************
<class 'distutils.unixccompiler.UnixCCompiler'>
preprocessor  = ['gcc', '-E']
compiler      = ['gcc', '-Wno-unused-result', '-Wsign-compare', '-Wunreachable-code', '-DNDEBUG', '-g', '-fwrapv', '-O3', '-Wall', '-Wstrict-prototypes', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64']
compiler_so   = ['gcc', '-Wno-unused-result', '-Wsign-compare', '-Wunreachable-code', '-DNDEBUG', '-g', '-fwrapv', '-O3', '-Wall', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64']
compiler_cxx  = ['g++']
linker_so     = ['gcc', '-bundle', '-undefined', 'dynamic_lookup', '-Wl,-rpath,/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-headerpad_max_install_names', '-arch', 'x86_64', '-Wl,-rpath,/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-headerpad_max_install_names', '-arch', 'x86_64']
linker_exe    = ['gcc']
archiver      = ['ar', 'rc']
ranlib        = ['ranlib']
libraries     = []
library_dirs  = ['/Users/kappamaki/anaconda/lib']
include_dirs  = ['/Users/kappamaki/anaconda/include/python3.6m']
********************************************************************************
building 'a' extension
compiling C++ sources
C compiler: g++ -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Users/kappamaki/anaconda/include -arch x86_64 -I/Users/kappamaki/anaconda/include -arch x86_64

creating /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var
creating /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders
creating /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders/r9
creating /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn
creating /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T
compile options: '-DUSE_GMP -DBOOST_SIMD_CONFIG_SUPPORT_POSIX_MEMALIGN -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_BLAS -I/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran -I/Users/kappamaki/anaconda/lib/python3.6/site-packages/numpy/core/include -I/Users/kappamaki/anaconda/include -I/Users/kappamaki/anaconda/lib/python3.6/site-packages/numpy/core/include -I/Users/kappamaki/anaconda/include/python3.6m -c'
extra options: '-std=c++11 -fno-math-errno'
g++: /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.cpp
exec_command(['g++', '-Wno-unused-result', '-Wsign-compare', '-Wunreachable-code', '-DNDEBUG', '-g', '-fwrapv', '-O3', '-Wall', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64', '-I/Users/kappamaki/anaconda/include', '-arch', 'x86_64', '-DUSE_GMP', '-DBOOST_SIMD_CONFIG_SUPPORT_POSIX_MEMALIGN', '-DENABLE_PYTHON_MODULE', '-D__PYTHRAN__=3', '-DPYTHRAN_BLAS_BLAS', '-I/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran', '-I/Users/kappamaki/anaconda/lib/python3.6/site-packages/numpy/core/include', '-I/Users/kappamaki/anaconda/include', '-I/Users/kappamaki/anaconda/lib/python3.6/site-packages/numpy/core/include', '-I/Users/kappamaki/anaconda/include/python3.6m', '-c', '/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.cpp', '-o', '/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.o', '-std=c++11', '-fno-math-errno'],)
Retaining cwd: /Users/kappamaki/Desktop/pythran_testing
_preserve_environment([])
_update_environment(...)
/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.cpp:105:17: warning: unknown attribute 'externally_visible' ignored [-Wunknown-attributes]
__attribute__ ((externally_visible))
                ^
In file included from /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.cpp:9:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/include/__builtin__/None.hpp:4:
In file included from /Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/types/NoneType.hpp:7:
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/__builtin__/id.hpp:26:8: warning: unused function 'id' [-Wunused-function]
  long id(long const &t)
       ^
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/__builtin__/id.hpp:31:8: warning: unused function 'id' [-Wunused-function]
  long id(double const &t)
       ^
/Users/kappamaki/anaconda/lib/python3.6/site-packages/pythran/pythonic/__builtin__/id.hpp:36:8: warning: unused function 'id' [-Wunused-function]
  long id(bool const &t)
       ^
4 warnings generated.
_update_environment(...)
g++ -bundle -undefined dynamic_lookup -Wl,-rpath,/Users/kappamaki/anaconda/lib -L/Users/kappamaki/anaconda/lib -headerpad_max_install_names -Wl,-rpath,/Users/kappamaki/anaconda/lib -L/Users/kappamaki/anaconda/lib -headerpad_max_install_names -arch x86_64 /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.o -L/Users/kappamaki/anaconda/lib -L/Users/kappamaki/anaconda/lib -lgmp -lgmpxx -lcblas -lblas -o /var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpornjjftg/a.cpython-36m-darwin.so
exec_command(['g++', '-bundle', '-undefined', 'dynamic_lookup', '-Wl,-rpath,/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-headerpad_max_install_names', '-Wl,-rpath,/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-headerpad_max_install_names', '-arch', 'x86_64', '/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmp8hmcwlxx/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpzfrgymzt.o', '-L/Users/kappamaki/anaconda/lib', '-L/Users/kappamaki/anaconda/lib', '-lgmp', '-lgmpxx', '-lcblas', '-lblas', '-o', '/var/folders/r9/s1k2nd110xzfxyfym7jmyb8r0000gn/T/tmpornjjftg/a.cpython-36m-darwin.so'],)
Retaining cwd: /Users/kappamaki/Desktop/pythran_testing
_preserve_environment([])
_update_environment(...)
_update_environment(...)
INFO: Generated module: a
INFO: Output: /Users/kappamaki/Desktop/pythran_testing/a.so
```

Any clues?

Thanks for your help.
Florian

--- a/pythran/pythran-darwin.cfg
+++ b/pythran/pythran-darwin.cfg
@@ -4,7 +4,7 @@ undefs=
 include_dirs=
 libs=gmp gmpxx
 library_dirs=
-cflags=-std=c++11 -fno-math-errno
+cflags=-std=c++11 -fno-math-errno -w
 ldflags=
 blas=blas
 CC=

